lib/java/com/google/android/material/lists/oneline/SingleLineItemAdapter.java

```diff
@@ -0,0 +1,31 @@
+package com.google.android.material.lists.oneline;
+
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView.ViewHolder;
+
+import com.google.android.material.lists.MaterialListAdapter;
+import com.google.android.material.lists.MaterialListBinder;
+
+/** An Adapter that shows single line list items */
+public class SingleLineItemAdapter extends MaterialListAdapter {
```

Comment:
What do you think about making `MaterialListAdapter` generic where you pass in the type of the view holder, rather than creating a new adapter for each type.

Reply:
Maybe this isn't necessary.. It would make `onCreateViewHolder()` more complicated.. you'd have to pass in some kind of factory or something to create instances of the correct type. This is probably just as simple.